api/repositories/service_instance_repository_test.go

```diff
@@ -1016,6 +1016,77 @@ var _ = Describe("ServiceInstanceRepository", func() {
 		})
 	})
 
+	Describe("GetServiceInstanceCredentials", func() {
+		var (
+			instanceGUID string
+			secretName   string
+			credential   map[string]any
+			secret       *corev1.Secret
+			getErr       error
+		)
+
+		BeforeEach(func() {
+			instanceGUID = prefixedGUID("service-instance")
+			secretName = prefixedGUID("secret")
+
+			serviceInstance := &korifiv1alpha1.CFServiceInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      instanceGUID,
+					Namespace: space.Name,
+				},
+				Spec: korifiv1alpha1.CFServiceInstanceSpec{
+					SecretName: secretName,
+					Type:       "user-provided",
+					Tags:       []string{"database", "mysql"},
+				},
+			}
+			Expect(k8sClient.Create(ctx, serviceInstance)).To(Succeed())
+
+			secretData, decodeErr := tools.ToCredentialsSecretData(map[string]any{"foo": "bar"})
+			Expect(decodeErr).ToNot(HaveOccurred())
+
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: space.Name,
+				},
+				Data: secretData,
+			}
+			Expect(k8sClient.Create(ctx, secret)).To(Succeed())
+		})
+
+		JustBeforeEach(func() {
+			credential, getErr = serviceInstanceRepo.GetServiceInstanceCredentials(ctx, authInfo, instanceGUID)
+		})
+
+		When("there are not permissions on the secret", func() {
+			It("returns a forbidden error", func() {
+				Expect(errors.As(getErr, &apierrors.ForbiddenError{})).To(BeTrue())
+			})
+		})
+
+		When("the user has permissions to get the secret", func() {
+			BeforeEach(func() {
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
+			})
+
+			It("returns the correct secret", func() {
+				Expect(getErr).ToNot(HaveOccurred())
+				Expect(credential).To(HaveKeyWithValue("foo", "bar"))
+			})
+
+			When("the service instance does not exist", func() {
+				BeforeEach(func() {
+					instanceGUID = "does-not-exist"
+				})
+
+				It("returns a 404 error", func() {
+					Expect(errors.As(getErr, &apierrors.NotFoundError{})).To(BeTrue())
+				})
+			})
+		})
+	})
```

Comment:
Could you also add a `When` to verify that the repository returns unprocessiable entity error when the secret data is invalid json?

Reply:
@danail-branekov Maybe this is a stupid question, but how would I achieve that ? I would have to create a secret with an invalid json format, how would I marshal it ?

Reply:
```
secret = &corev1.Secret{
				ObjectMeta: metav1.ObjectMeta{
					Name:      secretName,
					Namespace: space.Name,
				},
				Data: map[string][]byte{korifiv1alpha1.CredentialSecretKey, []byte{"not-json"}},
			}
Expect(k8sClient.Create(ctx, secret)).To(Succeed())
```

How about that?